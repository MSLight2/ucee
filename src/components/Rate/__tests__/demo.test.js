import { demoTest } from '../../../tests/shared/demoTest';
import RateDemo from '../demo';

demoTest('Rate', RateDemo);