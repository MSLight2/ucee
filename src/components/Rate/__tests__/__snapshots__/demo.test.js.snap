// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate demo test 1`] = `
Array [
  <p>
    基础用法
  </p>,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <br />,
  <br />,
  <p>
    只读
  </p>,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <br />,
  <br />,
  <p>
    禁用
  </p>,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item ucee-rate--disabled"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--disabled"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--disabled"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--disabled"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--disabled"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <br />,
  <br />,
  <p>
    半星
  </p>,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--active ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "100%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--active ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "100%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--active ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "100%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "40%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item ucee-rate--readOnly"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--half"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "50%%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
  </div>,
  "只读 ",
  <br />,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--active ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "100%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--active ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "100%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--active ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "100%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--half ucee-rate--active"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "50%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
      <div
        className="ucee-rate__star ucee-rate--half"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "width": "50%%",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "26px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <br />,
  <br />,
  <p>
    自定义颜色
  </p>,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": "#ff5500",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": "#ff5500",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": "#ff5500",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": "#1989fa",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": "#1989fa",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <br />,
  <br />,
  <p>
    自定义图标
  </p>,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": "#ff5500",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": "#ff5500",
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-like-fill"
          style={Object {}}
        />
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <p>
          星
        </p>
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <p>
          星
        </p>
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <p>
          星
        </p>
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <p>
          星
        </p>
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <p>
          星
        </p>
      </div>
    </div>
  </div>,
  <br />,
  <br />,
  <p>
    onChange监听
  </p>,
  <div
    className="ucee-rate"
  >
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star ucee-rate--active"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-fill-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
    <div
      className="ucee-rate__item"
      onClick={[Function]}
    >
      <div
        className="ucee-rate__star"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        <i
          className="ucee-icon ucee-icon-favor-light"
          style={
            Object {
              "fontSize": "20px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <br />,
]
`;
